### PR TITLE
Restructuring the `JSONLDRecord` and `BibliographicController` for more structured parsing of electronic location/access fields

### LINKED ISSUE #365: JSON-LD identifier returns the wrong url
The `identifier` property in the JSON-LD returns the catalog url instead of the ARK url.

Example bib: 2086336

https://github.com/pulibrary/marc_liberation/blob/17405858b7a0008e6c64cab8d5f25aa531e1453f/app/models/jsonld_record.rb#L105

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #365 by implementing `ElectronicLocation`, `ElectronicLocationFactory`, and modeling [electronic location subfield values](https://www.loc.gov/marc/bibliographic/bd856.html) in order to support this

### CODE REVIEW

File: app/services/electronic_locations_factory.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    values.map do |holding_id, url_values|
+      url_key = url_values.keys.first
+      ElectronicLocations::Holding.new(id: holding_id, value: url_key)
+    end
+  end
+
+  # Construct OnlineLocation Objects from Solr Document values
+  # @param values [Hash] Solr Document values for links Hash by URL
+  # @return [Array<OnlineLocation>]
+  def self.parse_online_locations(values)
```
Comment (eliotjordan) [inline app/services/electronic_locations_factory.rb]: This method doesn't seem to be used anywhere. Is it needed?

File: app/models/electronic_locations/online_location.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module ElectronicLocations
+  # Model for electronic location subfield values for online locations
+  class OnlineLocation
```
Comment (eliotjordan) [inline app/models/electronic_locations/online_location.rb]: `ElectronicLocations::OnlineLocation` doesn't seem to be used anywhere except as the super-class of `ElectronicLocations::IIIFManifestURI` and `ElectronicLocations::Holding`.  It's only method is over-ridden by both of it's sub-classes. Is it needed for something? Future expansion?

File: app/models/electronic_location.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  def initialize(access_method:, relationship:, subfields:, holdings:, iiif_manifest_uris:)
+    @access_method = access_method
+    @relationship = relationship
+    @subfields = subfields
+    index_holdings!(holdings)
+    @manifests = iiif_manifest_uris.map { |uri| uri.to_s }
+  end
+
+  # Return all sub-fields containing an ARK URL
+  # @return [Array]
```
Comment (eliotjordan) [inline app/models/electronic_location.rb]: Returns a String, rather than an Array.

**Comment by jrgriffiniii** [comment]
This is prepared for another review.

**Comment by tampakis** [comment]
@jrgriffiniii can you rebase against the changes in https://github.com/pulibrary/marc_liberation/pull/368 ? The Coveralls check fails because lines of no longer used code have been removed.

**Comment by jrgriffiniii** [comment]
This has been rebased against db3cd2b318e0a50043ae3e0afc50b50a804af5e6, with test coverage increased for the new Models and Service

**Pull request merged**